File: frontend/src/types/service.ts
export interface ServiceDTO {
    id: string;
    tenantId: string;
    companyId: string;
    code: string;
    name: string;
    description?: string;
    unitPrice?: number;
    costCenterId?: string;
    isActive: boolean;
    notes?: string;
    createdAt: string;
    createdBy: string;
    updatedAt?: string;
    updatedBy?: string;
}

export interface CreateServiceDTO {
    tenantId: string;
    companyId: string;
    code: string;
    name: string;
    description?: string;
    unitPrice?: number;
    costCenterId?: string;
    isActive?: boolean;
    notes?: string;
}

export interface UpdateServiceDTO {
    code?: string;
    name?: string;
    description?: string;
    unitPrice?: number;
    costCenterId?: string;
    isActive?: boolean;
    notes?: string;
}

export interface ServiceFilters {
    companyId?: string;
    activeOnly?: boolean;
    searchText?: string;
}

File: backend/src/main/java/br/com/sigeve/sigeve_prodution/model/Service.java
package br.com.sigeve.sigeve_prodution.model;

import jakarta.persistence.*;
import lombok.*;

import java.math.BigDecimal;
import java.util.UUID;

/**
 * Entidade Service - Catálogo de Serviços
 * Representa tipos de serviços que podem ser contratados (ex: Costura, Lavagem, etc.)
 */
@Entity
@Table(name = "tab_services")
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Builder
public class Service extends AuditFull {

    @Id
    @GeneratedValue(strategy = GenerationType.AUTO)
    @Column(name = "service_id", updatable = false, nullable = false)
    private UUID id;

    @Column(name = "tenant_id", nullable = false)
    private UUID tenantId;

    @Column(name = "company_id", nullable = false)
    private UUID companyId;

    @Column(name = "code", nullable = false, length = 50)
    private String code;

    @Column(name = "name", nullable = false, length = 200)
    private String name;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "unit_price", precision = 15, scale = 2)
    private BigDecimal unitPrice;

    @Column(name = "cost_center_id")
    private UUID costCenterId;

    @Column(name = "is_active", nullable = false)
    private Boolean isActive = true;

    @Column(name = "notes", columnDefinition = "TEXT")
    private String notes;
}
